custom/icds_reports/tasks.py

```diff
@@ -1636,7 +1636,10 @@ def reconcile_data_not_in_ucr(reconciliation_status_pk):
     known_bad_doc_ids = doc_ids_in_pillow_error.intersection(invalid_doc_ids)
 
     # republish_kafka_changes
-    for doc_id, doc_subtype, sql_modified_on in data_not_in_ucr:
+    # running the data accessor again to avoid storing all doc ids in memory
+    # since run time is relatively short and does not scale with number of errors
+    # but the number of doc ids will increase with the number of errors
+    for doc_id, doc_subtype, sql_modified_on in get_data_not_in_ucr(status_record):
```

Comment:
this is a very expensive operation isn't it? seems like it would be better to just keep it in memory or else buffer to disk

Reply:
It runs in a minute or so, so it did not seem too bad, and the downside of finding the time when we actually need it (a day with a lot of failures) and that causing memory errors seemed worth the repeated db call, but happy to change it if you prefer the other one. Will merge it now to make sure today's task succeeds, but need to do followup here anyway so can make that change in the followup if necessary